src/coreclr/src/vm/interoplibinterface.cpp

```diff
@@ -655,63 +652,64 @@ namespace
             if (gc.objRef == NULL)
             {
                 gc.objRef = CallGetObject(&gc.implRef, identity, flags);
-                if (gc.objRef == NULL)
-                    COMPlusThrow(kArgumentNullException);
             }
 
-            // Construct the new context with the object details.
-            DWORD flags = (resultHolder.Result.FromTrackerRuntime
-                            ? ExternalObjectContext::Flags_ReferenceTracker
-                            : ExternalObjectContext::Flags_None) |
-                          (uniqueInstance
-                            ? ExternalObjectContext::Flags_None
-                            : ExternalObjectContext::Flags_InCache);
-            ExternalObjectContext::Construct(
-                resultHolder.GetContext(),
-                identity,
-                GetCurrentCtxCookie(),
-                gc.objRef->GetSyncBlockIndex(),
-                flags);
-
-            if (uniqueInstance)
+            if (gc.objRef != NULL)
```

Comment:
>if (gc.objRef != NULL) [](start = 12, length = 22)

Please add a comment why this is okay. Something about we are handling the error cases in managed for other scenarios or the like.